src/app/features/location/address-search/address-search.component.ts

```diff
@@ -88,10 +90,17 @@ export class AddressSearchComponent implements OnInit {
   /** do not display selected item in the input field because this should be an empty search field */
   displayFn = () => "";
 
-  constructor(private location: GeoService) {}
+  constructor(
+    private location: GeoService,
+    private dialog: MatDialog,
+  ) {}
 
   ngOnInit() {
     this.initSearchPipeline();
+    // Keep a local array for isInputInOptions
+    this.filteredOptions.subscribe((options) => {
+      this.filteredOptionsArray = options;
+    });
```

Comment:
I think we could simplify this by using BehaviorSubject instead of Subject as the type of `filteredOptions`: https://www.learnrxjs.io/learn-rxjs/subjects/behaviorsubject

it then has a property `filteredOptions.value` without having to create this ourselves